theano/tensor/nnet/abstract_conv.py

```diff
@@ -301,6 +301,12 @@ def grad(self, inp, grads):
                                                self.filter_flip)(
 
             bottom, top, weights.shape[-2:])
+
+        # Make sure that the broadcastable pattern of the inputs is used
+        # for the gradients, even if the grad opts are not able to infer
+        # that the dimensions are broadcastable.
+        d_bottom = patternbroadcast(d_bottom, bottom.broadcastable)
```

Comment:
Actually, I don't think that's right: `d_bottom` is created by `_gradInputs`, which does not get passed `bottom`. Similarly, `d_weights` is created by `_gradWeights`, which does not have the broadcastable pattern of `weights`.
The only place where we have all that information is here, in the `grad` method.